pkg/client/cache/buildconfig.go

```diff
@@ -0,0 +1,100 @@
+package cache
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/labels"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+// StoreToBuildConfigLister gives a store List and Exists methods. The store must contain only buildconfigs.
+type StoreToBuildConfigLister interface {
+	List() ([]*buildapi.BuildConfig, error)
+	GetConfigsForImageStreamTrigger(*imageapi.ImageStream) ([]*buildapi.BuildConfig, error)
+}
+
+// StoreToBuildConfigListerImpl implements a StoreToBuildConfigLister
+type StoreToBuildConfigListerImpl struct {
+	cache.Indexer
+}
+
+// List all buildconfigs in the store.
+func (s *StoreToBuildConfigListerImpl) List() ([]*buildapi.BuildConfig, error) {
+	configs := []*buildapi.BuildConfig{}
+	for _, c := range s.Indexer.List() {
+		configs = append(configs, c.(*buildapi.BuildConfig))
+	}
+	return configs, nil
+}
+
+// GetConfigsForImageStream returns all the build configs that are triggered by the provided image stream
+// by searching through using the ImageStreamReferenceIndex (build configs are indexed in the cache
+// by image stream references).
+func (s *StoreToBuildConfigListerImpl) GetConfigsForImageStreamTrigger(stream *imageapi.ImageStream) ([]*buildapi.BuildConfig, error) {
```

Comment:
I don't love taking an image stream here - it would be better to take a stream namespace and name.